tests/mmm-ext-fulfill.spec.ts

```diff
@@ -0,0 +1,925 @@
+
+      const {
+        mint: extraMint,
+        recipientTokenAccount: extraRecipientTokenAccount,
+      } = await createTestMintAndTokenT22VanillaExt(
+        connection,
+        wallet.payer,
+        seller.publicKey,
+        groupAddress,
+      );
+
+      const { key: extraNftSellState } = getMMMSellStatePDA(
+        program.programId,
+        poolData.poolKey,
+        extraMint,
+      );
+      const extraPoolAta = await getAssociatedTokenAddress(
+        extraMint,
+        poolData.poolKey,
+        true,
+        TOKEN_2022_PROGRAM_ID,
+      );
+
+      let [initReferralBalance, initSellerBalance, initBuyerBalance] =
+        await Promise.all([
+          connection.getBalance(poolData.referral.publicKey),
+          connection.getBalance(seller.publicKey),
+          connection.getBalance(buyer.publicKey),
+        ]);
+
+      const expectedLpFees = LAMPORTS_PER_SOL * 0.02;
+      const expectedTakerFees = LAMPORTS_PER_SOL * 0.01;
+      const expectedBuyPrices = getSolFulfillBuyPrices({
+        totalPriceLamports: LAMPORTS_PER_SOL,
+        lpFeeBp: 200,
+        takerFeeBp: 100,
+        metadataRoyaltyBp: 0,
+        buysideCreatorRoyaltyBp: 0,
+        makerFeeBp: 0,
+      });
+
+      const tx = await program.methods
+        .extFulfillBuy({
+          assetAmount: new anchor.BN(1),
+          minPaymentAmount: expectedBuyPrices.sellerReceives,
+          allowlistAux: null,
+          takerFeeBp: 100,
+          makerFeeBp: 0,
+        })
+        .accountsStrict({
+          payer: seller.publicKey,
+          owner: wallet.publicKey,
+          cosigner: cosigner.publicKey,
+          referral: poolData.referral.publicKey,
+          pool: poolData.poolKey,
+          buysideSolEscrowAccount: solEscrowKey,
+          assetMint: extraMint,
+          payerAssetAccount: extraRecipientTokenAccount,
+          sellsideEscrowTokenAccount: extraPoolAta,
+          ownerTokenAccount: extraRecipientTokenAccount,
+          allowlistAuxAccount: SystemProgram.programId,
+          systemProgram: SystemProgram.programId,
+          tokenProgram: TOKEN_2022_PROGRAM_ID,
+          associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
+          rent: SYSVAR_RENT_PUBKEY,
+          sellState: extraNftSellState,
+        })
+        .transaction();
+
+      const blockhashData = await connection.getLatestBlockhash();
+      tx.feePayer = seller.publicKey;
+      tx.recentBlockhash = blockhashData.blockhash;
+      tx.partialSign(cosigner, seller);
+
+      await sendAndAssertTx(connection, tx, blockhashData, false);
+
+      let tokenAccountRent =
+        (await getTokenAccountRent(connection)) +
+        IMMUTABLE_OWNER_EXTENSION_LAMPORTS;
+      const sellStatePDARent = await getSellStatePDARent(connection);
+
+      const expectedTxFees = SIGNATURE_FEE_LAMPORTS * 2; // cosigner + payer
+      {
+        const [sellerBalance, referralBalance] = await Promise.all([
+          connection.getBalance(seller.publicKey),
```

Comment:
can we also assert that the nft was recieved by the correct address (I think pool in this case)